src/java.base/share/classes/java/io/SimpleIO.java

```diff
@@ -0,0 +1,961 @@
+ * questions.
+ */
+
+package java.io;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class provides easy to use console and file I/O methods for simple applications.
+ * <p>
+ * The console I/O methods allow interaction with the user through the keyboard and console/screen.
+ * The method {@link #input(String)} accepts input from the keyboard
+ * while allowing the user to edit the input with arrows and backspaces, as well as
+ * allowing the user to scroll through previously entered input. The methods {@link #inputInt(String)},
+ * {@link #inputLong(String)}, {@link #inputFloat(String)} and {@link #inputDouble(String)}
+ * can be used to input primitive values. The methods {@link #print(Object...)} and
+ * {@link #printLine(Object...)} allow developers to display data directly to the console/screen.
+ * For example:
+ * {@snippet lang="java":
+ * import static java.io.SimpleIO.*;
+ *
+ * public class Example1 {
+ *     public static void main(String[] args) {
+ *         String name = input("Enter name>> ");
+ *         printLine("Hello", name);
+ *     }
+ * }
+ * }
+ * The file I/O method methods are designed to work with UTF-8 text files. The developer can
+ * specify the file by either a simple file name string or by using a {@link Path}.
+ * The developer also has the choice of working with the whole content as a string or
+ * as a string list representing lines of content.
+ * <p>
+ * File I/O is futher simplifed by throwing {@link UncheckedIOException} for caught
+ * {@link IOException IOExceptions}. This means that the developer does not require
+ * wrapping simple I/O invocations in try blocks. For example:
+ * {@snippet lang="java":
+ * import static java.io.SimpleIO.*;
+ *
+ * public class Example2 {
+ *     public static void main(String[] args) {
+ *         String content = read("original.txt");
+ *         write("copy.txt", content);
+ *     }
+ * }
+ * }
+ * If the developer chooses to handle exceptions they can add the try block to catch
+ * the {@link UncheckedIOException}. For example:
+ * {@snippet lang="java":
+ * import static java.io.SimpleIO.*;
+ *
+ * public class Example3 {
+ *     public static void main(String[] args) {
+ *         try {
+ *           String content = read("original.txt");
+ *           write("copy.txt", content);
+ *         } catch (UncheckedIOException ex) {
+ *              printLine("File not copied");
+ *         }
+ *     }
+ * }
+ * }
+ * @since  21
+ */
+public final class SimpleIO {
+    /**
+     * Private constructor.
+     */
+    private SimpleIO() {
+        throw new AssertionError("private constructor");
```

Comment:
Wouldn't it be to more reasonable to throw `UnsupportedOperationException` here?

Reply:
In his 'Effective Java' book, Joshua Bloch suggests using `AssertionError` because it sends a clear message that the exception should not be caught.